### DISCUSSION TITLE
Decrease the DBs' size

### DISCUSSION THREAD

**Discussion by John-Gee** [DeFiCh/app#753] [Ideas]
This is related to my other discussion in https://github.com/DeFiCh/app/issues/745 .

**Is your feature request related to a problem? Please describe.**
It's not great having to download AND store all the data for DeFiCh, and assuming the network takes off, it's only going to get worse.

**Describe the solution you'd like**
Take a similar approach the other networks did, allow pruning on the DBs, and of course allow to start that way!
I think Monero keeps some percentage of the data it should not need, to still have a safer network, please correct me if I'm wrong, but I think that's a sensible way to go at it.

**Describe alternatives you've considered**
The compression of the DBs described in the link above, but even at a unlikely 10% ratio, it would still grow too much too quickly. The 2 solutions would  work nicely together though.

Of course, there may be a much smarter way to go at this problem; I believe I recently read about a new crypto network that supposedly never really grows its blockchain' data as it keeps shrinking it every new block, but I forgot the name of that. Hopefully someone has a clue what I'm alluding to.

**Additional context**
Even if the light wallets are delivered soon, I believe we should still strive to have a standard decentralized wallet as light as possible, in all resources.

**Comment by thomasfriebel** [comment]
Piggybacking this request:
After running the wallet for less than 1 month, it already filled my MacBook's disk quite significantly.#
<img width="441" alt="image" src="https://user-images.githubusercontent.com/3184165/116309777-78077b80-a7a9-11eb-9044-3f6d83e7d7b8.png">
Deleted the snapshot.zip to gain some room back, but can only be a matter of a few weeks until disk is stuffed again thanks to DFI wallet.

**Reply by John-Gee** [reply]
So I think the team is working on something to make it use less space, and later they'll also look at compression. Not sure if the first one will be enough for you or if the 2nd one will be soon enough..

**Comment by John-Gee** [comment]
You guys did amazing!

If things are as they should on my other post, the storage problem is solved, if you can do the same with the downloaded data, at least in the snapshot form, this suggestion can definitely be closed.

It does make me wonder though if with pruning (maybe you do that already?), compression, and whatever else I'm not thinking of, you can make light wallets not even needed. edit: ok you're already pruning it so that's why.

**Reply by John-Gee** [reply]
Downloads are also pruned so it's all much smaller now!
